compiler/src/dotty/tools/dotc/printing/SyntaxHighlighting.scala

```diff
@@ -355,4 +365,89 @@ object SyntaxHighlighting {
 
     newBuf.toIterable
   }
+
+  private class NoReporter extends Reporter {
+    override def doReport(m: MessageContainer)(implicit ctx: Context): Unit = ()
+  }
+
+  private val ignoredKwds = Set(nme.ARROWkw, nme.EQ, nme.EQL, nme.COLONkw)
+
+  def highlight(in: String)(ctx0: Context): String = {
+    import dotty.tools.dotc.ast.untpd._
+
+    implicit val ctx: Context = ctx0.fresh.setReporter(new NoReporter)
+
+    val source = new SourceFile("<highlighting>", in.toCharArray)
+    val parser = new Parser(source)
+    val trees = parser.blockStatSeq()
+
+    val colorAt = Array.fill(in.length)(NoColor)
+
+    def highlightRange(from: Int, to: Int, color: String) = {
+      try {
+        for (i <- from until to)
+          colorAt(i) = color
+      } catch {
+        case _: IndexOutOfBoundsException =>
+          ctx.error("Encountered tree with invalid position, please open an issue with the code snippet that caused the error")
```

Comment:
`ctx.error` will not print anything since your are using a dummy reporter. Simply use `println`